server/apps/immich/src/api-v1/album/album.service.ts

```diff
@@ -174,4 +196,40 @@ export class AlbumService {
       album.albumThumbnailAssetId = dto.albumThumbnailAssetId || null;
     }
   }
+
+  async createAlbumSharedLink(authUser: AuthUserDto, dto: CreateAlbumShareLinkDto): Promise<SharedLinkResponseDto> {
+    const album = await this._getAlbum({ authUser, albumId: dto.albumId });
+
+    const sharedLink = await this.shareCore.createSharedLink(authUser.id, {
+      sharedType: SharedLinkType.ALBUM,
+      expiredAt: dto.expiredAt,
+      allowUpload: dto.allowUpload,
+      album: album,
+      assets: album.assets?.map((asset) => asset.assetInfo) || [],
+      description: dto.description,
+    });
+
+    return mapSharedLinkToResponseDto(sharedLink);
+  }
```

Comment:
You are importing the dto from the share controller to use here. I think we should just consolidate all the share related function (create, read, update, delete) into that service instead, and maybe just use this service to check album validation/access before creating the link.